src/CsvHelper/Configuration/ConfigurationFunctions.cs

```diff
@@ -16,35 +16,36 @@ namespace CsvHelper.Configuration
 	public static class ConfigurationFunctions
 	{
 		private static readonly char[] lineEndingChars = new char[] { '\r', '\n' };
+		private static readonly string[] lineEndings = new[] { "\r\n", "\r", "\n" };
 
 		/// <summary>
 		/// Throws a <see cref="ValidationException"/> if <see name="HeaderValidatedArgs.InvalidHeaders"/> is not empty.
 		/// </summary>
 		public static void HeaderValidated(HeaderValidatedArgs args)
 		{
-			if (args.InvalidHeaders.Count() == 0)
+			if (args.InvalidHeaders.Length == 0)
 			{
 				return;
 			}
 
 			var errorMessage = new StringBuilder();
 			foreach (var invalidHeader in args.InvalidHeaders)
 			{
-				errorMessage.AppendLine($"Header with name '{string.Join("' or '", invalidHeader.Names)}'[{invalidHeader.Index}] was not found.");
+				errorMessage.Append("Header with name '").Append(string.Join("' or '", invalidHeader.Names)).Append("'[").Append(invalidHeader.Index).AppendLine("] was not found.");
 			}
 
 			if (args.Context.Reader.HeaderRecord != null)
 			{
 				foreach (var header in args.Context.Reader.HeaderRecord)
 				{
-					errorMessage.AppendLine($"Headers: '{string.Join("', '", args.Context.Reader.HeaderRecord)}'");
+					errorMessage.Append("Headers: '").Append(string.Join("', '", args.Context.Reader.HeaderRecord)).AppendLine("'");
```

Comment:
You can perhaps use `AppendJoin` on the stringbuilder directly here. It's available from Netstandard2.1 onwards